lib/ansible/modules/cloud/amazon/aws_secret.py

```diff
@@ -0,0 +1,358 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, REY Remi
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: aws_secret
+short_description: Manage secrets stored in AWS Secrets Manager.
+description:
+    - Create, update, and delete secrets stored in AWS Secrets Manager.
+author: "REY Remi (@rrey)"
+version_added: "2.8"
+requirements: [ 'botocore>=1.10.0', 'boto3' ]
+options:
+  name:
+    description:
+    - Friendly name for the secret you are creating.
+    required: true
+  state:
+    description:
+    - Whether the secret should be exist or not.
+    default: 'present'
+    choices: ['present', 'absent']
+  recovery_window:
+    description:
+    - Only used if state is absent.
+    - Specifies the number of days that Secrets Manager waits before it can delete the secret.
+    - If set to 0, the deletion is forced without recoery.
+    default: 30
+  description:
+    description:
+    - Specifies a user-provided description of the secret.
+  kms_key_id:
+    description:
+    - Specifies the ARN or alias of the AWS KMS customer master key (CMK) to be
+      used to encrypt the `secret_string` or `secret_binary` values in the versions stored in this secret.
+  secret_type:
+    description:
+    - Specifies the type of data that you want to encrypt.
+    choices: ['binary', 'string']
+    default: 'string'
+  secret:
+    description:
+    - Specifies string or binary data that you want to encrypt and store in the new version of the secret.
+    default: ""
+  tags:
+    description:
+    - Specifies a list of user-defined tags that are attached to the secret.
+  rotation_lambda:
+    description:
+    - Specifies the ARN of the Lambda function that can rotate the secret.
+  rotation_interval:
+    description:
+    - Specifies the number of days between automatic scheduled rotations of the secret.
+    default: 30
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+
+EXAMPLES = r'''
+- name: Add string to AWS Secrets Manager
+  aws_secret:
+    name: 'test_secret_string'
+    state: present
+    secret_type: 'string'
+    secret: "{{ super_secret_string }}"
+
+- name: remove string from AWS Secrets Manager
+  aws_secret:
+    name: 'test_secret_string'
+    state: absent
+    secret_type: 'string'
+    secret: "{{ super_secret_string }}"
+'''
+
+
+RETURN = r'''
+secret:
```

Comment:
It will help out people a lot if you can document the detailed contents of the return. You can do this pretty easily by running the module, saving the returned json into a file, and then running https://github.com/ansible/ansible/blob/devel/hacking/return_skeleton_generator.py with the file.

Reply:
Thanks for the tips ! awesome !